### PR TITLE
PR: Preserve file permissions when saving in the editor

### LINKED ISSUE #9381: Overwritten file permissions on save
## Description
If one saves a file it a) cannot follow my symlink and b) removes my file permissions such that it's not executable anymore. In other words I think on each save the original file is deleted and replaced by a new one

### What steps will reproduce the problem?
Click save

## Versions

* Spyder version: 4.0.0b2 
* Python version: 3.6.7
* Qt version: 5.12.3
* PyQt5 version: 5.12.2
* Operating System: Linux 4.15.0-50-generic

### Dependencies

```
pygments >=2.0     :  2.4.0 (OK)
qdarkstyle >=2.6.4 :  2.6.8 (OK)
sphinx >=0.6.6     :  2.0.1 (OK)
psutil >=0.3       :  5.6.2 (OK)
pyls >=0.19.0;<0.25:  0.24.0 (OK)
nbconvert >=4.0    :  5.5.0 (OK)
pandas >=0.13.1    :  0.23.4 (OK)
numpy >=1.7        :  1.16.3 (OK)
sympy >=0.7.3      :  1.1.1 (OK)
cython >=0.21      :  0.26.1 (OK)
qtconsole >=4.2.0  :  4.4.4 (OK)
IPython >=4.0      :  7.5.0 (OK)
matplotlib >=2.0.0 :  2.2.2 (OK)
pylint >=0.25      :  2.3.1 (OK)
```

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Preserve file permission when saving with `atomic_writes`

### Issue(s) Resolved

Fixes #9381 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

File: spyder/utils/encoding.py [hunk 1]
```diff
@@ -235,10 +235,25 @@ def write(text, filename, encoding='utf-8', mode='wb'):
         with open(filename, mode) as textfile:
             textfile.write(text)
     else:
+        # Based in the solution at untitaker/python-atomicwrites#42
+        # Needed to fix file permissions overwritting
+        # See spyder-ide/spyder#9381
+        try:
+            FileNotFoundError
+        except NameError:  # Python 2
+            FileNotFoundError = OSError
```
Comment (ccordoba12) [inline spyder/utils/encoding.py]: @dalthviz, this is very convoluted. Let's use `OSError` for now and let's leave a comment below saying that we should really use `FileNotFoundError` when we move to Python 3.

File: spyder/utils/encoding.py [hunk 2]
```diff
@@ -235,10 +235,21 @@ def write(text, filename, encoding='utf-8', mode='wb'):
         with open(filename, mode) as textfile:
             textfile.write(text)
     else:
+        # Based in the solution at untitaker/python-atomicwrites#42
+        # Needed to fix file permissions overwritting
+        # See spyder-ide/spyder#9381
+        try:
+            original_mode = os.stat(filename).st_mode
+        except OSError:
```
Comment (ccordoba12) [inline spyder/utils/encoding.py]: Please add an inline comment in this same line like this

    except OSError: # Change to FileNotFoundError for PY3

**Comment by daniel-edler** [comment]
Thanks a lot

**Comment by Khalilsqu** [comment]
> Thanks a lot

do not forget the 10 euro bounty! 😉

**Pull request merged**